### PR TITLE
Skip handoff event on unloaded partition

### PR DESCRIPTION
**Pull request by weiliu1031**
Signed-off-by: Wei Liu <wei.liu@zilliz.com>
Related to #20305 
/kind bug

### CODE REVIEW

**Comment by weiliu1031** [comment]
rerun ut

File: internal/querycoordv2/observers/handoff_observer.go [hunk 1]
```diff
@@ -272,6 +291,43 @@ func (ob *HandoffObserver) tryHandoff(ctx context.Context, segment *querypb.Segm
...
+			}, retry.Attempts(5))
+
+			if err != nil {
+				// collection has been dropped or released
+				if strings.Contains(err.Error(), "CollectionNotExists") ||
+					ob.meta.GetCollection(segment.GetCollectionID()) == nil {
+					return false
+				}
+				// collection not released , but can get partition list to check handoff
+				panic("handoff check load status failed due to get partitions failed")
```
Comment (czs007) [inline internal/querycoordv2/observers/handoff_observer.go:321]: suggest to  just return false, and do not panic.

File: internal/querycoordv2/observers/handoff_observer.go [hunk 2]
```diff
@@ -272,6 +291,43 @@ func (ob *HandoffObserver) tryHandoff(ctx context.Context, segment *querypb.Segm
 	}
 }
 
+func (ob *HandoffObserver) checkLoadStatus(segment *querypb.SegmentInfo) bool {
+	if ob.meta.GetCollection(segment.GetCollectionID()) != nil {
+		// if collection is loaded, should check whether the partition has been droped!
+		if ob.loadedPartitions[segment.GetSegmentID()] == nil {
```
Comment (yah01) [inline internal/querycoordv2/observers/handoff_observer.go:297]: The key should be `CollectionID`, not `SegmentID`

**Comment by weiliu1031** [comment]
rerun ut

**Comment by congqixia** [comment]
@weiliu1031 could you please set the PR title to a meaningful one when you fix the unit test? thanks!

**Comment by weiliu1031** [comment]
> @weiliu1031 could you please set the PR title to a meaningful one when you fix the unit test? thanks!

ok

**Review by yah01 — CHANGES_REQUESTED** [review]

**Comment by weiliu1031** [comment]
rerun ut

File: internal/querycoordv2/observers/handoff_observer.go [hunk 3]
```diff
@@ -100,10 +108,23 @@ func (ob *HandoffObserver) Register(collectionIDs ...int64) {
...
 	for _, collectionID := range collectionIDs {
 		delete(ob.collectionStatus, collectionID)
+		delete(ob.handoffSubmitOrders, collectionID)
 	}
+
+	collectionSet := typeutil.NewUniqueSet(collectionIDs...)
+	filteredEvents := make(map[int64]*HandoffEvent, len(ob.handoffEvents))
+	for segmentID, event := range ob.handoffEvents {
+		if !collectionSet.Contain(event.Segment.GetCollectionID()) {
+			filteredEvents[segmentID] = event
```
Comment (yah01) [inline internal/querycoordv2/observers/handoff_observer.go:123]: just `delete(ob.handoffEvents, segmentID)`

File: internal/querycoordv2/observers/handoff_observer.go [hunk 4]
```diff
@@ -115,6 +136,13 @@ func (ob *HandoffObserver) StartHandoff(collectionIDs ...int64) {
 	}
 }
 
+func (ob *HandoffObserver) getHandoffEventNum() int {
```
Comment (yah01) [inline internal/querycoordv2/observers/handoff_observer.go:139]: Any method accessed outside should be public, `getHandoffEventNum` -> `EventNum` or `GetEventNum`, the type name has implied they are handoff event

Comment (weiliu1031) [reply internal/querycoordv2/observers/handoff_observer.go:139]: just for ut, no need to expose to outside.

**Pull request merged**